eng/test-determinism.ps1

```diff
@@ -0,0 +1,395 @@
+[CmdletBinding(PositionalBinding = $false)]
+param([string]$configuration = "Debug",
+  [string]$msbuildEngine = "vs",
+  [string]$altRootDrive = "q:",
+  [switch]$help,
+  [switch]$norestore,
+  [switch]$rebuild)
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function Print-Usage() {
+  Write-Host "Usage: test-determinism.ps1"
+  Write-Host "  -configuration <value>    Build configuration ('Debug' or 'Release')"
+  Write-Host "  -msbuildEngine <value>    Msbuild engine to use to run build ('dotnet', 'vs', or unspecified)."
+  Write-Host "  -bootstrapDir             Directory containing the bootstrap compiler"
+  Write-Host "  -altRootDrive             The drive we build on (via subst) for verifying pathmap implementation"
+}
+
+if ($help) {
+  Print-Usage
+  exit 0
+}
+
+# List of binary names that should be skipped because they have a known issue that
+# makes them non-deterministic.
+$script:skipList = @()
+function Run-Build([string]$rootDir, [string]$logFileName) {
+
+  # Clean out the previous run
+  Write-Host "Cleaning binaries in $rootDir"
+  $binDir = Get-BinDir $rootDir
+  $objDir = Get-ObjDir $rootDir
+  $stopWatch = [System.Diagnostics.StopWatch]::StartNew()
+  Write-Host "Cleaning binaries in $binDir"
+  Remove-Item -Recurse $binDir -ErrorAction SilentlyContinue
+  Write-Host "Cleaning binaries in $objDir"
+  Remove-Item -Recurse $objDir -ErrorAction SilentlyContinue
+  $stopWatch.Stop()
+  Write-Host "Cleaning took $($stopWatch.Elapsed)"
+
+  $solution = Join-Path $rootDir (Join-Path "service" "FSharp.Compiler.Service.sln")
+
+  if ($logFileName -eq "") {
+    $logFileName = [IO.Path]::GetFileNameWithoutExtension($projectFilePath)
+  }
+  $logFileName = [IO.Path]::ChangeExtension($logFileName, ".binlog")
+  $logFilePath = Join-Path $LogDir $logFileName
+
+  Stop-Processes
+
+  Write-Host "Building $solution using $bootstrapDir"
+  MSBuild $toolsetBuildProj `
+    /p:Configuration=$configuration `
+    /p:Projects=$solution `
+    /p:RepoRoot=$rootDir `
+    /p:Restore=true `
+    /p:Build=true `
+    /p:Rebuild=false `
+    /p:Pack=false `
+    /p:Sign=false `
+    /p:SignType=Test `
```

Comment:
Since you're manually specifying the sign type here, you should be able to delete the `_SignType` variable from the `.yml` file and you could even remove the PR condition; if this can reliably run in the signed build I'd feel more comfortable about determinism not breaking.

Reply:
I'll delete the `_SignType`, I'm not confident enough that it won't break in `main` again (I may have missed some parameters?), is there a reliable way running it the same way we do in `main`, but from this branch?